### PR TITLE
OCLOMRS-262: Fix all the errors/warnings that appear in the browser console in development mode

### PR DESCRIPTION
**Pull request by muhozi**
# JIRA TICKET NAME:

[Fix all the errors/warnings that appear in the browser console in development mode](https://issues.openmrs.org/browse/OCLOMRS-262)

# Summary:
In development mode, there are warnings/errors that appear in the browser console. Some code refactoring is required to fix those warnings and errors. Most of those errors and warnings are related to prop types validation and DOM nesting.

Those errors need to be fixed to leave the development console error/warning free.

### CODE REVIEW

File: src/components/bulkConcepts/component/PreviewCard.jsx [hunk 1]
```diff
@@ -44,11 +44,12 @@ PreviewCard.propTypes = {
   closeModal: PropTypes.func.isRequired,
   addConcept: PropTypes.func.isRequired,
   concept: PropTypes.shape({
-    display_name: PropTypes.string.isRequired,
-    descriptions: PropTypes.array.isRequired,
```
Comment (dessHub) [inline src/components/bulkConcepts/component/PreviewCard.jsx]: Arrays take a shape describing what it contains. Check out this [tread](https://stackoverflow.com/questions/32325912/react-proptype-array-with-shape)

Comment (dessHub) [inline src/components/bulkConcepts/component/PreviewCard.jsx]: Arrays take a shape describing what it contains. Check out this [tread](https://stackoverflow.com/questions/32325912/react-proptype-array-with-shape)

Comment (dessHub) [inline src/components/bulkConcepts/component/PreviewCard.jsx]: Arrays take a shape describing what it contains. Check out this [tread](https://stackoverflow.com/questions/32325912/react-proptype-array-with-shape)

Comment (dessHub) [inline src/components/bulkConcepts/component/PreviewCard.jsx]: Arrays take a shape describing what it contains. Check out this [tread](https://stackoverflow.com/questions/32325912/react-proptype-array-with-shape)

Comment (muhozi) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: @dessHub Thanks for pointing this out. I think that I'll create a ticket for implementing props validation properly. The scope of this ticket was only to remove console warnings and errors during development mode.

Comment (Duncanian) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: @dessHub , according to React 16.5.2, it is the correct way of doing it. [Check this out](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes)

Comment (dessHub) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: It should be `PropTypes.arrayOf()`

Comment (dessHub) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: Then its fine

Comment (dessHub) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: Sorry for the tread

Comment (muhozi) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: @dessHub I think that you use `PropTypes.arrayOf()` if it is an array of objects.

File: src/components/dashboard/components/dictionary/DictionaryDetailCard.jsx [hunk 2]
```diff
@@ -61,11 +59,11 @@ const DictionaryDetailCard = (props) => {
   return (
     <div className="dictionaryDetails">
       <section className="backLink">
-        <Link
-          to="/home"
-          className="backLinkText"
-        >
-          <i className="fas fa-arrow-left"> <span className="span-text">Go Back Home</span></i>
+        <Link to="/home" className="backLinkText">
```
Comment (Daramola98) [inline src/components/dashboard/components/dictionary/DictionaryDetailCard.jsx]: Good job on this PR, but I think it is a best practice to name CSS classes without camelCase

Comment (Daramola98) [reply src/components/dashboard/components/dictionary/DictionaryDetailCard.jsx]: Something like this would be more appropriate back-link-text

**Comment by dondrzzy** [comment]
Hey @muhozi, nice work but best case scenario would have been, 1 PR for every file where you need to correct ESlint errors.

File: src/components/dashboard/components/dictionary/DictionaryDetailCard.jsx (hunk 2 above)
Comment (muhozi) [reply src/components/dashboard/components/dictionary/DictionaryDetailCard.jsx]: @Daramola98 Thanks for pointing this out, I think that we should create a ticket for this. Because this is not related to this ticket.

File: src/components/login/container/index.jsx [hunk 3]
```diff
@@ -96,18 +93,19 @@ export class Login extends Component {
...
+                    <br />
                     <a
                       href="https://qa.openconceptlab.org"
                       target="_blank"
                       rel="noopener noreferrer"
                     >
-                    Go to Traditional OCL
-                    </a><br />
-                    <Link to="/loginDetails" >Having Trouble with Login?</Link>
+                      Go to Traditional OCL
```
Comment (dkayiwa) [inline src/components/login/container/index.jsx]: Is this change also related to the ticket?

File: src/components/login/container/index.jsx [hunk 4]
```diff
@@ -49,10 +49,8 @@ export class Login extends Component {
         <Title title="Login" />
         <div className="row form-container">
           <div className="col-lg-5 col-md-9 col-sm-12">
-            <p className="paragraphs">
-              <h1 className="login-title">Open Concept Lab</h1>
```
Comment (dkayiwa) [inline src/components/login/container/index.jsx]: Is this change also related to the ticket?

File: src/components/login/container/index.jsx [hunk 5]
```diff
@@ -13,17 +13,6 @@ export class Login extends Component {
     username: '',
   };
 
-  handleInput = (event) => {
-    const { name, value } = event.target;
-    this.setState({ [name]: value });
```
Comment (dkayiwa) [inline src/components/login/container/index.jsx]: Is this change also related to the ticket?

File: src/components/dashboard/components/dictionary/DictionaryVersionsTable.jsx [hunk 6]
```diff
@@ -18,11 +18,13 @@ const DictionaryVersionsTable = (version) => {
       <td>{id}</td>
       <td>{ (new Date(updated_on)).toLocaleDateString('en-US', DATE_OPTIONS)}</td>
       <td><a href={version_url}>Browse in OCL</a> <Link className="downloadConcepts" onClick={download} to="#">Download</Link> <Link className="subscription-link" onClick={() => { showSubModal(version_url); }} to="#">Subscription URL</Link></td>
-      <SubscriptionModal
-        show={show}
```
Comment (dkayiwa) [inline src/components/dashboard/components/dictionary/DictionaryVersionsTable.jsx]: Is this change also related to the ticket?

File: src/components/dashboard/components/dictionary/DictionaryDetailCard.jsx [hunk 7]
```diff
@@ -147,44 +145,67 @@ const DictionaryDetailCard = (props) => {
...
           <fieldset>
             <legend>Released Version</legend>
             <div className="card" id="versionCard">
               <table>
-                <tr>
-                  <th>Version</th>
-                  <th>Date</th>
-                  <th>Actions</th>
-                </tr>
-                {releasedVersions.length >= 1 ? (
```
Comment (dkayiwa) [inline src/components/dashboard/components/dictionary/DictionaryDetailCard.jsx]: Is this change also related to the ticket?

**Comment by dkayiwa** [comment]
@muhozi the commit message should include the ticket number as per https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by muhozi** [comment]
> @muhozi the commit message should include the ticket number as per [wiki.openmrs.org/display/docs/Pull+Request+Tips](https://wiki.openmrs.org/display/docs/Pull+Request+Tips)

@dkayiwa Thanks for pointing this out, I changed the commit message.
